modules/apps/portal-vulcan/portal-vulcan-test/src/testIntegration/java/com/liferay/portal/vulcan/graphql/dto/test/GraphQLDTOContributorTest.java

```diff
@@ -0,0 +1,424 @@
+
+		Assert.assertEquals(
+			"test",
+			JSONUtil.getValueAsString(
+				jsonObject, "JSONObject/data", "JSONObject/createGraphQLDTO",
+				"Object/name"));
+	}
+
+	@Test
+	public void testGraphQLDTOContributorDeleteDTO() throws Exception {
+		_testGraphQLDTOContributor.addDTO(1, "test");
+
+		GraphQLField graphQLField = new GraphQLField(
+			"mutation",
+			new GraphQLField(
+				"deleteGraphQLDTO",
+				HashMapBuilder.<String, Object>put(
+					"id", 1
+				).build()));
+
+		JSONObject jsonObject = _invoke(graphQLField);
+
+		Assert.assertTrue(
+			JSONUtil.getValueAsBoolean(
+				jsonObject, "JSONObject/data", "Object/deleteGraphQLDTO"));
+
+		jsonObject = _invoke(graphQLField);
+
+		Assert.assertFalse(
+			JSONUtil.getValueAsBoolean(
+				jsonObject, "JSONObject/data", "Object/deleteGraphQLDTO"));
+	}
+
+	@Test
+	public void testGraphQLDTOContributorGetDTO() throws Exception {
+		_testGraphQLDTOContributor.addDTO(1, "test");
+
+		JSONObject jsonObject = _invoke(
+			new GraphQLField(
+				"query",
+				new GraphQLField(
+					"graphQLDTO",
+					HashMapBuilder.<String, Object>put(
+						"id", 1
+					).build(),
+					new GraphQLField("id"), new GraphQLField("name"))));
+
+		Assert.assertEquals(
+			"test",
+			JSONUtil.getValueAsString(
+				jsonObject, "JSONObject/data", "JSONObject/graphQLDTO",
+				"Object/name"));
+	}
+
+	@Test
+	public void testGraphQLDTOContributorGetDTOs() throws Exception {
+		_testGraphQLDTOContributor.addDTO(1, "test1");
+		_testGraphQLDTOContributor.addDTO(2, "test2");
+
+		JSONObject jsonObject = _invoke(
+			new GraphQLField(
+				"query",
+				new GraphQLField(
+					"graphQLDTOs", Collections.emptyMap(),
+					new GraphQLField("totalCount"),
+					new GraphQLField("items", new GraphQLField("name")))));
+
+		Assert.assertEquals(
+			2,
+			JSONUtil.getValueAsLong(
+				jsonObject, "JSONObject/data", "JSONObject/graphQLDTOs",
+				"Object/totalCount"));
+
+		Assert.assertEquals(
+			"test1",
+			JSONUtil.getValueAsString(
+				jsonObject, "JSONObject/data", "JSONObject/graphQLDTOs",
+				"JSONArray/items", "Object/0", "Object/name"));
+
+		Assert.assertEquals(
+			"test2",
+			JSONUtil.getValueAsString(
+				jsonObject, "JSONObject/data", "JSONObject/graphQLDTOs",
+				"JSONArray/items", "Object/1", "Object/name"));
+	}
+
+	@Test
+	public void testGraphQLDTOContributorUpdateDTO() throws Exception {
+		_testGraphQLDTOContributor.addDTO(1, "test");
+
+		JSONObject jsonObject = _invoke(
+			new GraphQLField(
+				"query",
+				new GraphQLField(
+					"graphQLDTO",
+					HashMapBuilder.<String, Object>put(
+						_testGraphQLDTOContributor.getIdName(), 1
+					).build(),
+					new GraphQLField("id"), new GraphQLField("name"))));
+
+		Assert.assertEquals(
+			"test",
+			JSONUtil.getValueAsString(
+				jsonObject, "JSONObject/data", "JSONObject/graphQLDTO",
+				"Object/name"));
+	}
+
+	public static class GraphQLDTO {
+
+		public GraphQLDTO() {
+		}
+
+		public GraphQLDTO(long id, String name) {
```

Comment:
If you are using this just for a test, make fields public/protected and avoid getters